tests/Dfe.EarlyYearsQualification.UnitTests/Attributes/RedirectIfDateMissingAttributeTests.cs

```diff
@@ -0,0 +1,52 @@
+using Dfe.EarlyYearsQualification.Web.Attributes;
+using Dfe.EarlyYearsQualification.Web.Models;
+using Dfe.EarlyYearsQualification.Web.Services.UserJourneyCookieService;
+using FluentAssertions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Routing;
+using Moq;
+
+namespace Dfe.EarlyYearsQualification.UnitTests.Attributes;
+
+[TestClass]
+public class RedirectIfDateMissingAttributeTests
+{
+    [TestMethod]
+    public void OnAuthorization_NoDateInCookie_RedirectsToDateSelectionRoute()
+    {
+        var userJourneyCookieService = new Mock<IUserJourneyCookieService>();
+        userJourneyCookieService.Setup(x => x.GetUserJourneyModelFromCookie()).Returns(new UserJourneyModel());
+        
+        var filter = new RedirectIfDateMissingAttribute.RedirectIfDateMissingFilter(userJourneyCookieService.Object);
+        
+        var actionContext = new ActionContext(new DefaultHttpContext(), new RouteData(), new ActionDescriptor());
+        var actionExecutingContext = new ActionExecutingContext(actionContext, new List<IFilterMetadata> { filter }, new Dictionary<string, object?>(), null);
```

Comment:
If the inbound parameter isn't nullable, you can easily sort these issues by passing in a `new object()` rather than `null`.

Reply:
Ah, I see you've done that while I was writing my comment! 🙂